### ISSUE TITLE
may i read-only the markdown article?

### ISSUE THREAD

**Issue by xp44mm** [typora/typora-issues#710]
when i read some article that is myself or everybody. i don't want to change anything i don't want to change.
read-only state just like static web page. it don't changed.

**Comment by orschiro** [comment]
Can't you just read it without doing any changes? I mean, you don't have to edit the file, do you?

**Comment by xp44mm** [comment]
to do so target is **To prevent unintentional modifications**
for example, i read Typora Quick Start. this kind of article don't need to be changed and shouldn't be changed. 
some other soft, for example, pdf reader. just read not write. pdf pro can write. it's target is protect document don't be changed.
excel has lock function to protect worksheet. and so on.

**Comment by shesgottadevelopit** [comment]
I found this program searching for a solution for just viewing/reading markdown files.  I have a lot of markdown files for class and I just want to read them and reference them, not make any changes unintentionally. So I'd love it if that was a feature to allowed us to turn read-only on/off.

**Comment by abnerlee** [comment]
On macOS you could just "lock" the file.
For Window/Linux, just click 'Do not save' when you close the window. 

So this is a low priority task in my own opinion, since there are some "alternative" solution. 
We will consider this, but no plans for near future

**Comment by zhengpd** [comment]
@abnerlee Maybe just add a shortcut key to toggle the "lock" of file on macOS. That could be more convenient.

**Comment by vassudanagunta** [comment]
This is really a duplicate of #89.  If you are in favor of this one, I recommend you go vote for #89.

**Comment by vassudanagunta** [comment]
@abnerlee 

> On macOS you could just "lock" the file.

Read-only mode should do more than prevent editing. One should be able to click on links without having to hold down the command key to navigate them. 

Maybe mark this as a duplicate of #89?

**Comment by abnerlee** [comment]
Merge into #89

**Comment by abnerlee** [comment]
I think is issue is about for reading/previewing purpose

#89 is for presentation/demo purpose

**Comment by vassudanagunta** [comment]
Ok, then I'll move my **viewer mode** comment from #89 to here:

A read-only mode would be very useful. It allows Typora to be both a Markdown **editor** and a Markdown ***viewer***. 

### In viewer mode:

- **Clicking links** navigate to the target rather then rather then exposes the link markup for editing.
- **Optimized for reading**. Markups is not exposed.
- You can't accidentally modify the content, which is a big deal since on OSes like macOS changes are automatically saved.
- Optionally, a different theme/CSS would be used.  

@abnerlee Maybe change the title of this Issue to either "Read-only Mode", "Viewer Mode" or "Reader Mode"?

**Comment by kymowang** [comment]
viewer mode is really useful and a nice feature of the soft.  I don't have to export to html or pdf format for getting good experience of reading.

**Comment by alshdavid** [comment]
I too would appreciate this feature. 

I distribute the documentation of my projects in their respective repos, I use a lot of hyperlinks between documents to tie it together. 

A reading mode that allows people to follow hyperlinks without holding cnt/cmd.

I would also require clicked hyperlinks to open in the same window.

These two things are holding me back from being able to recommend Typora to my team as the default documentation reading tool

**Comment by lovelock** [comment]
> Ok, then I'll move my **viewer mode** comment from #89 to here:
> 
> A read-only mode would be very useful. It allows Typora to be both a Markdown **editor** and a Markdown _**viewer**_.
> ### In viewer mode:
> 
>     * **Clicking links** navigate to the target rather then rather then exposes the link markup for editing.
> 
>     * **Optimized for reading**. Markups is not exposed.
> 
>     * You can't accidentally modify the content, which is a big deal since on OSes like macOS changes are automatically saved.
> 
>     * Optionally, a different theme/CSS would be used.
> 
> 
> @abnerlee Maybe change the title of this Issue to either "Read-only Mode", "Viewer Mode" or "Reader Mode"?

For me, as a programmer I write a lot of code in md files, when I want to copy code from codeblocks it is annoying because spaces before and after the codeblock would always be selected. If View mode is on, then the issue is gone.

**Comment by wilbura009** [comment]
On Linux, you can `chmod -w yourFile.md` and then when you open the file in typora you cannot modify it. I do this, I hope it helps.

**Comment by felixzwettler** [comment]
I have a similar use-case for typora as @alshdavid and would also greatly appreciate this feature.

**Comment by guesswh0** [comment]
I see there is already a checkbox option "Locked" to lock/unlock file on top of the application frame. I find it enough.

**Comment by undersound** [comment]
@guesswh0 can't seem to find this option. Could you provide a screendump or such?  On linux here btw.

**Comment by juniorxxue** [comment]
<img width="1127" alt="Screenshot 2021-07-08 at 4 06 07 PM" src="https://user-images.githubusercontent.com/31076337/124885983-6989db80-e006-11eb-813f-1a5d9aec3135.png">
@undersound click the title and then there's a lock option appearing in drop-down menu

**Comment by undersound** [comment]
@juniorxxue  you mean the title of the file that is displayed in the GUI window? Not seeing that option unfortuanetely. 
Perhaps this is a mac only option?

**Comment by XCanG** [comment]
I vote for this one too.

_I didn't saw this feature request, but I mentioned about it in other issue_ https://github.com/typora/typora-issues/issues/4645#issue-988375178

**Comment by AhmedThahir** [comment]
Yes please! Everyone would really benefit from this feature.

**Comment by joolz** [comment]
+1

**Comment by cpasoft** [comment]
> A read-only mode would be very useful. It allows Typora to be both a Markdown **editor** and a Markdown _**viewer**_.
> ### In viewer mode:
> 
>     * **Clicking links** navigate to the target rather then rather then exposes the link markup for editing.
> 
>     * **Optimized for reading**. Markups is not exposed.
> 
>     * You can't accidentally modify the content, which is a big deal since on OSes like macOS changes are automatically saved.
> 
>     * Optionally, a different theme/CSS would be used.

YES, please.. This suggestion is a MUST. An only viewer mode is necessary

**Comment by alanalvarado** [comment]
That's exactly why I never bought typora, and just abandoned evernote.
I want my notes to be WIKI style. And almost nobody supports that simple feature. At least Obsidian does.

**Comment by edisuc** [comment]
+1

**Comment by edisuc** [comment]
I also vote for the feature "Viewer mode". There should be an option to activate it with a command line parameter.

**Comment by soto97** [comment]
-1 

Typora is a markdown editor, at least that's how I see it. I would really rather not have modes in Typora, like an editor mode or viewer mode. That will start to make Typora feel more like Vim, which I would rather avoid.

If you want a markdown viewer, I suggest using Marked 2. That's all that app does. Then you can use Marked 2 for viewing and Typora for editing.

**Comment by bc0n** [comment]
+1 
I'd also like to be able to toggle a viewer mode, where I can edit the markdown elsewhere and have typora render and display.
Paid user on linux mint, fwiw.

**Comment by eyebrowkang** [comment]
+1
Sometimes I have completed my editing work, but I can't close it because I need to read it. In this case, I don't want my operations which out of my attention mess up my file, so the viewer mode is useful, it is like reader mode in Jetbrains.

**Comment by gorishabh** [comment]
Team typora, Any updates on this issue? Can you please include this feature in the latest release this has been pending for some time now and lots of other folks have already raised this above?

**Even writing this comment has a `Preview` tab where I am actually verifying this formatting**  and you can understand the importance of this feature and why _we have paid for this software_ to have better response from the team

Please look into this 🙏🏽

**Comment by jphbutler64** [comment]
+1
Essential to avoid corrupting files by mistake!  E.g. typing in wrong window etc, etc.
Especially since the layout is so good looking that I don't want to read markdown in anything else.

**Comment by MEntOMANdo** [comment]
+1 
This is a must. I personally love this program, but can't purchase it without this feature for my use case.

The whole point of markdown files is to make the content easy to read by yourself and others. When reading that content, you don't want to make changes. 

As OP mentioned, even the Typora Quick Start would benefit from this. The developer had to make a file called "DO NOT ADD FILES HERE" in the Quick Start, just to prevent people from modifying this content!  Not to mention, even casually clicking around might cause an edit, and then MacOS automatically saves the file. The result is a modification with no warning at all, and no audit trail since this is not a version control system or database.

Clicking links - even in the Quick Start (!!!!) doesn't work. Expecting someone to even know that they must Ctrl+Click is ridiculous. There are many other examples listed above: math, copying code blocks, etc.

A solution is just to present the md file in read-only mode each and every time, and provide a Pencil icon somewhere. Click on the pencil to unlock/edit the document. Similar to how a Wiki works. You could provide a preferences item that says: "Open files in [Editing |v|] mode" (choices are Editing, Reading) which would allow someone to configure it to match today's behavior.

**Comment by vugar005** [comment]
+ 1 from me too. Main reason from my part is similar: **To prevent unintentional modifications** while reading and scrolling through file.

**Comment by zarthcode** [comment]
Just dropping a note, I read *piles* of markdown files, and rarely write/edit them.  Ctrl+Click is non-starter when reading generated api docs, wanting to copy/paste, and NOT ACCIDENTALLY ALTER ANYTHING!

I'm passing on the buying a license because I want an offline reader w/occasional edit.

**Comment by MEntOMANdo** [comment]
> an offline reader w/occasional edit

It's funny because on the site, the page title (visible in the browser tab) is "Typora - a markdown editor, markdown reader." So specifically calls out 2 modes of operation, but there is only 1 mode - editor.

The first words on the website are "Readable and Writeable" - but again, it's only just a live editor.

And then he's adamant that no one wants to view docs as read-only, even though the product can't even display its own tutorial. Very stubborn. 

Typora - just provide us the OPTION to view docs in read-only mode until we click a pencil to edit it.  Why should we use TWO programs to read and edit?

**Comment by g4x86** [comment]
I second this feature request. More than 90% of my working documents were written in Markdown by Typora, so as time goes on I frequently go back to view those documents but no need to modify them. As Typora already supports many different themes, it's quite natural to find a comfortable theme and use it to view Markdown documents in a read-only mode without accidentally changing the documents. So far, I have to export every Markdown document to PDF format: just for viewing (not printing) it! I hope Typora can add this feature soon. (by the way, I have 3 Typora licenses)

**Comment by MEntOMANdo** [comment]
Developer arrogance (on any project) is frustrating. Paying customers request a fairly simple feature that's optional and doesn't change the experience unless you enable it, and it gets ignored for product-philosophy reasons. As a former product manager, I empathize with the tough decision. But I can also say that if you disregard feedback from paying customers, it doesn't work out well.

**Comment by CX330Blake** [comment]
+1 on this please. Please hear from us, your users.

**Comment by mrbrownjeremy** [comment]
+1 this should be a View option. It not only helps prevent unintentional input, but it's also nice to navigate a doc without the markdown characters revealing themselves (which affects layout and can interrupt reading).

**Comment by EdgarDegas** [comment]
This issue was created in 2017 and today here we are now in a vibe coding era. 

I want just to READ WHAT MY AI AGENT WROTE and MAKE NO CHANGE to it.

**Comment by SdotVdot** [comment]
Is it already implemented?

**Comment by mg** [comment]
Is this feature on the schedule? I use Neovim to edit my markdown files, and am only looking for a good markdown viewer. A command line parameter that I could use when I open Typora from Neovim that would instruct Typora to be in read only mode is something that would make this program way more useful for me.

**Comment by omega32** [comment]
Same here. I just found out about Typora, and I love it. It is already my favorite MD viewer.

But I would really, really love a read-only mode.
Bonus points if it could be activated with a `CLI` option.

**Comment by say8425** [comment]
Yes, that's right. Back in 2017 when this issue was raised, the priority must have been low. But now, we have to read all kinds of markdown files churned out by every AI under the sun. So I think this issue has now become much more important.

I'd like to ask the Typora team once again: is read-only mode on the roadmap?

**Comment by XCanG** [comment]
I think at this moment it would be easier to create your own Markdown editor/viewer. There is quite a lot of open source libraries that could be used to solve the hardest part: parsing Markdown and making HTML out of it. What would be left is only creating GUI for it. In the past year I make Markdown integration for a blog site with Markdown editor using [MDXEditor](https://mdxeditor.dev/) as a core. I could make desktop version of a similar viewer/editor if there is need for that.

**Comment by kites262** [comment]
Really need that. 😕

**Comment by MingoDynasty** [comment]
Seems crazy that a "read-only mode" does not exist, when it seems so trivial to implement, meanwhile other Markdown editors like Obsidian have it built-in, and Typora plugins prove it possible (https://github.com/obgnail/typora_plugin#read_only).
